### PR TITLE
disable Stealth of stealthsuit when in critical or death AND syndicate icon shows when in stealth.

### PR DESCRIPTION
**Pull request by jorgun**
## About the PR

makes the use of the syndicate stealth hardsuit visible when in crit or dead.

## Why / Balance
 
its a problem that nukes just dissapere when in critical and cant be found by teammates.

## Technical details

there is a problem: when you go critical, the component stops updating.
 that the stealth component dont start untill you take off the helmet and put it back on.

## Media

image

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- tweak: Steathsuit stops working when in crit
- tweak: Nuclear Oprative icon is visable when in stealth

### CODE REVIEW

File: Content.Shared/Stealth/SharedStealthSystem.cs [hunk 1]
```diff
@@ -61,17 +61,21 @@ public virtual void SetEnabled(EntityUid uid, bool value, StealthComponent? comp
...
+                component.Enabled = component.EnabledOnDeath;
+            else
+                component.Enabled = component.EnabledOnCrit;
         }
         else
         {
             component.Enabled = true;
         }
-
+        SetEnabled(uid, component.Enabled, component);// to update the sprite;
```
[the remark this answers is not in the document]
Comment (jorgun) [reply Content.Shared/Stealth/SharedStealthSystem.cs:78]: coderabbit, your drunk. delete this

**Comment by a-person5660** [comment]
a better solution to this is just making the nukie hud show when invisible.

**Comment by jorgun** [comment]
> a better solution to this is just making the nukie hud show when invisible.

I know. But as ther was already a function for make it visible on death. Not mutch was required to expand that for critical.

**Comment by jorgun** [comment]
only the icon becomes visable, there is no way of making hp bar visable whitout makeing it visable to crew aswell

**Pull request merged**